miniconf/src/node.rs

```diff
@@ -173,10 +173,45 @@ impl<T, const S: char> From<T> for Path<T, S> {
     }
 }
 
+/// String split/skip wrapper, smaller/simpler than `.split(S).skip(1)`
+#[derive(Copy, Clone, Default, Debug, PartialEq, PartialOrd, Serialize, Deserialize)]
+#[repr(transparent)]
+pub struct PathIter<'a, const S: char>(Option<&'a str>);
+
+impl<'a, const S: char> PathIter<'a, S> {
+    #[inline]
+    fn new(s: &'a str) -> Self {
+        let mut s = Self(Some(s));
+        // Skip the first part to disambiguate between
+        // the one-Key Keys `[""]` and the zero-Key Keys `[]`.
+        // This is relevant in the case of e.g. `Option` and newtypes.
+        // See the corresponding unittests (`just_option`).
+        s.next();
+        s
+    }
+}
+
+impl<'a, const S: char> Iterator for PathIter<'a, S> {
+    type Item = &'a str;
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        self.0.map(|s| {
+            let pos = s
+                .chars()
+                .map_while(|c| (c != S).then_some(c.len_utf8()))
```

Comment:
Would https://doc.rust-lang.org/core/iter/trait.Iterator.html#method.skip_while work for us? I.e. just skip the empty string?

Reply:
Ah I think the docs about about empty keys vs empty path for options above explains it.

Reply:
I think skip while empty would work but it's less strict than the  implicit definition above (a path element is always prefixed by a separator). And we'd have to be much kore careful w.r.t. composability in e.g. CLI, Mqtt, and serial settings where then we'd have to conditionally insert the separator if we allow path to start without.

Reply:
Yeah being strict here feels more right for sure. Thanks for humoring me